### PR TITLE
publish fatjar settings

### PR DESCRIPTION
**Pull request by TharinduDG**
- Publish `fatJar` 
- Based on https://github.com/cognitedata/cdp-spark-datasource/commit/50186afb81f99a16777ed35304d80e7dd52485f1 and `sbt-assembly` 1.x [documentation](https://github.com/sbt/sbt-assembly#publishing-not-recommended).

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -22,7 +22,7 @@ lazy val commonSettings = Seq(
   organization := "com.cognite.spark.datasource",
   organizationName := "Cognite",
   organizationHomepage := Some(url("https://cognite.com")),
-  version := "1.4.54",
+  version := "1.4.55.SNAPSHOT.1",
```
Comment (wjoel) [inline build.sbt:25]: Use -SNAPSHOT to be able to publish snapshots to Sonatype

File: build.sbt [hunk 2]
```diff
@@ -141,6 +131,21 @@ lazy val library = (project in file("."))
...
+    assemblyMergeStrategy := {
+      case PathList("META-INF", xs@_*) => MergeStrategy.discard
+      case _ => MergeStrategy.first
+    },
+    assemblyShadeRules := {
+      val shadePackage = "cognite.spark.v1.shaded"
+      Seq(
+        ShadeRule.rename("shapeless.**" -> s"$shadePackage.shapeless.@1").inAll,
+        ShadeRule.rename("cats.**" -> s"$shadePackage.cats.@1").inAll,
+        ShadeRule.rename("cats.kernel.**" -> s"$shadePackage.cats.kernel.@1").inAll,
```
Comment (wjoel) [inline build.sbt:143]: I don't think we need this, should be covered by the previous wildcard rule?

File: build.sbt [hunk 3]
```diff
@@ -141,6 +131,21 @@ lazy val library = (project in file("."))
...
       "org.log4s" %% "log4s" % log4sVersion
     ),
+    assemblyMergeStrategy := {
+      case PathList("META-INF", xs@_*) => MergeStrategy.discard
+      case _ => MergeStrategy.first
+    },
+    assemblyShadeRules := {
+      val shadePackage = "cognite.spark.v1.shaded"
+      Seq(
+        ShadeRule.rename("shapeless.**" -> s"$shadePackage.shapeless.@1").inAll,
```
Comment (wjoel) [inline build.sbt:141]: The old commit also had `ShadeRule.rename("com.google.protobuf.**" -> "repackaged.cognite_spark_datasource.com.google.protobuf.@1").inAll,`, is it not needed?

File: build.sbt (hunk 1 above)
Comment (TharinduDG) [reply build.sbt:25]: done.

File: build.sbt (hunk 3 above)
Comment (TharinduDG) [reply build.sbt:141]: I didn't add this since the issue we're having is with cats. Anyway I'll add this as well.

**Review by wjoel — APPROVED** [review]

**Pull request merged**